aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp

```diff
@@ -515,13 +516,19 @@ inline static void apply_svd_cusolver_gesvdj(const Tensor& A, const Tensor& U, c
       infos.data_ptr<int>() + i,
       gesvdj_params
     );
+
+    // The following code can be used to check or report the gesvdj residual.
+    // double residual = 0;
+    // TORCH_CUSOLVER_CHECK(cusolverDnXgesvdjGetResidual(handle, gesvdj_params, &residual));
+    // printf("gesvdj residual = %.6e\n", residual);
```

Comment:
Thought: Should we check that the residual is less than some constant, and if it isn't then report that `gesvdj` was not able to converge and default to a different backend?

Reply:
`cusolverDnXgesvdjGetResidual` does introduce device-host sync which may negatively affect performance. Also, there isn't a good boundary value that can effectively distinguish if an ill-conditioned matrix is converged. In my experience, something like `4e-9` could be used, but not always effective.